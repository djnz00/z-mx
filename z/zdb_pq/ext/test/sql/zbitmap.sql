SELECT ''::zbitmap;
SELECT zbitmap_length('');
SELECT zbitmap_send('');
SELECT ','::zbitmap;
SELECT '-'::zbitmap;
SELECT '0'::zbitmap;
SELECT '0,'::zbitmap;
SELECT '0-'::zbitmap;
SELECT ',0'::zbitmap;
SELECT zbitmap_get(',0', 0);
SELECT zbitmap_get(',0', 1);
SELECT '-0'::zbitmap;
SELECT zbitmap_clr_range('', 0, 0);
SELECT zbitmap_set_range('', 0, 0);
SELECT zbitmap_set_range('', 0, 1);
SELECT zbitmap_set_range('', 1, 2);
SELECT zbitmap_set_range('0-1,3-4,6,8-', 60, 68);
SELECT zbitmap_send(zbitmap_set_range('0-1,3-4,6,8-', 60, 68));
SELECT zbitmap_xor(zbitmap_set_range('0-1,3-4,6,8-', 60, 68), '1-4');
SELECT ''::zbitmap = ''::zbitmap;
