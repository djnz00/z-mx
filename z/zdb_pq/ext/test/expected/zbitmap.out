SELECT ''::zbitmap;
 zbitmap 
---------
 
(1 row)

SELECT zbitmap_length('');
 zbitmap_length 
----------------
 0
(1 row)

SELECT zbitmap_send('');
    zbitmap_send    
--------------------
 \x0000000000000000
(1 row)

SELECT ','::zbitmap;
 zbitmap 
---------
 
(1 row)

SELECT '-'::zbitmap;
ERROR:  invalid input syntax for zbitmap: "-"
LINE 1: SELECT '-'::zbitmap;
               ^
SELECT '0'::zbitmap;
 zbitmap 
---------
 0
(1 row)

SELECT '0,'::zbitmap;
 zbitmap 
---------
 0
(1 row)

SELECT '0-'::zbitmap;
 zbitmap 
---------
 0-
(1 row)

SELECT ',0'::zbitmap;
 zbitmap 
---------
 0
(1 row)

SELECT zbitmap_get(',0', 0);
 zbitmap_get 
-------------
 t
(1 row)

SELECT zbitmap_get(',0', 1);
 zbitmap_get 
-------------
 f
(1 row)

SELECT '-0'::zbitmap;
ERROR:  invalid input syntax for zbitmap: "-0"
LINE 1: SELECT '-0'::zbitmap;
               ^
SELECT zbitmap_clr_range('', 0, 0);
 zbitmap_clr_range 
-------------------
 
(1 row)

SELECT zbitmap_set_range('', 0, 0);
 zbitmap_set_range 
-------------------
 
(1 row)

SELECT zbitmap_set_range('', 0, 1);
 zbitmap_set_range 
-------------------
 0
(1 row)

SELECT zbitmap_set_range('', 1, 2);
 zbitmap_set_range 
-------------------
 1
(1 row)

SELECT zbitmap_set_range('0-1,3-4,6,8-', 60, 68);
 zbitmap_set_range 
-------------------
 0-1,3-4,6,8-68
(1 row)

SELECT zbitmap_send(zbitmap_set_range('0-1,3-4,6,8-', 60, 68));
                    zbitmap_send                    
----------------------------------------------------
 \x0000000000000002ffffffffffffff5b000000000000001f
(1 row)

SELECT zbitmap_xor(zbitmap_set_range('0-1,3-4,6,8-', 60, 68), '1-4');
 zbitmap_xor 
-------------
 0,2,6,8-68
(1 row)

SELECT ''::zbitmap = ''::zbitmap;
 ?column? 
----------
 t
(1 row)

